### PR TITLE
tests(spread): make release test less flaky

### LINKED ISSUE #1362: CI: `charm-upload-and-release` test is flaky
### Bug Description

The charm upload test seems to be fairly frequently failing due to the charm store reporting the release as having updated immediately after publish.

Likely we just need to sleep for a few seconds and maybe do some retries to prevent the test from getting the revision before the values have propagated.

### To Reproduce

`spread google:ubuntu-22.04-64:tests/spread/store/charm-upload-and-release`

See: https://github.com/canonical/charmcraft/actions/runs/6644905285/job/18055442120?pr=1352

### Environment

CI

### charmcraft.yaml

```shell
N/A
```

### Relevant log output

```shell
++ charmcraft upload gh-ci-charmcraft-charm_ubuntu-22.04-amd64.charm --format=json
Starting to push 'gh-ci-charmcraft-charm_ubuntu-22.04-amd64.charm'
Uploading... (--->)
Uploading... (<---)
Uploading bytes ended, id 00bbc7f6-1081-45a1-8ee0-e45f381ccabc
Upload 00bbc7f6-1081-45a1-8ee0-e45f381ccabc started, got status url /v1/charm/gh-ci-charmcraft-charm/revisions/review?upload-id=00bbc7f6-1081-45a1-8ee0-e45f381ccabc
Status checked: {'revisions': [{'errors': None, 'revision': None, 'status': 'new', 'upload-id': '00bbc7f6-1081-45a1-8ee0-e45f381ccabc'}]}
Status checked: {'revisions': [{'errors': None, 'revision': 1025, 'status': 'scanned', 'upload-id': '00bbc7f6-1081-45a1-8ee0-e45f381ccabc'}]}
Status checked: {'revisions': [{'errors': None, 'revision': 1025, 'status': 'approved', 'upload-id': '00bbc7f6-1081-45a1-8ee0-e45f381ccabc'}]}
+ uploaded_revno=1025
++ jq -r --arg revno 1025 '.[] | select(.revision|tostring==$revno)'
++ charmcraft revisions gh-ci-charmcraft-charm --format=json
+ up_revision='{
  "revision": 1025,
  "version": "1025",
  "created_at": "2023-10-25T19:15:16Z",
  "status": "approved"
}'
++ jq -r .created_at
++ echo '{' '"revision":' 1025, '"version":' '"1025",' '"created_at":' '"2023-10-25T19:15:16Z",' '"status":' '"approved"' '}'
+ up_revision_created=2023-10-25T19:15:16Z
+ [[ 2023-10-25T19:12:13+00:00 < 2023-10-25T19:15:16Z ]]
+ charmcraft release gh-ci-charmcraft-charm -r 1025 -c edge
Revision 1025 of 'gh-ci-charmcraft-charm' released to edge
++ jq -r '.[] | select(.track=="latest") | .mappings[0].releases | .[] | select(.channel=="latest/edge")'
++ charmcraft status gh-ci-charmcraft-charm --format=json
+ edge_release='{
  "status": "open",
  "channel": "latest/edge",
  "version": "1024",
  "revision": 1024,
  "resources": [
    {
      "name": "example-file",
      "revision": 1
    },
    {
      "name": "example-image",
      "revision": 1
    }
  ],
  "expires_at": null
}'
++ jq -r .revision
++ echo '{' '"status":' '"open",' '"channel":' '"latest/edge",' '"version":' '"1024",' '"revision":' 1024, '"resources":' '[' '{' '"name":' '"example-file",' '"revision":' 1 '},' '{' '"name":' '"example-image",' '"revision":' 1 '}' '],' '"expires_at":' null '}'
+ edge_revision=1024
+ test 1024 -ge 1025
```

### PR DESCRIPTION
**Pull request by lengau**
Fixes #1362

### CODE REVIEW

**Review by syu-w — COMMENTED** [review]
Missing a test that failed to release after the loop?

**Comment by lengau** [comment]
@syu-w good catch, fixed!

**Review by syu-w — APPROVED** [review]

File: tests/spread/store/charm-upload-and-release/task.yaml [hunk 1]
```diff
@@ -42,10 +42,20 @@ execute: |
...
-  # validate the channel map
-  edge_release=$(charmcraft status $CHARM_DEFAULT_NAME --format=json | jq -r '.[] | select(.track=="latest") | .mappings[0].releases | .[] | select(.channel=="latest/edge")')
-  edge_revision=$(echo $edge_release | jq -r .revision)
-  # check that the current release greater or equal than what was
-  # uploaded, because other tests running concurrently may have
-  # also released (but for sure cannot be a previous revision)
-  test $edge_revision -ge $uploaded_revno
+  # Releases may have some delay in the craft store, so sleep for a bit before trying and try up to 10 times.
+  for i in {1..10}
+  do
```
Comment (tigarmo) [inline tests/spread/store/charm-upload-and-release/task.yaml:47]: should we have a max number of tries here to keep this from looping forever?

Comment (lengau) [reply tests/spread/store/charm-upload-and-release/task.yaml:47]: This only tries up to 10 times.

Comment (tigarmo) [reply tests/spread/store/charm-upload-and-release/task.yaml:47]: image

**Review by tigarmo — APPROVED** [review]

**Pull request merged**